### ISSUE TITLE
Keyboard shortcuts

### ISSUE THREAD

**Issue by nikolockenvitz** [nikolockenvitz/2048#7]
- shortcut for full-screen and normal mode (e.g. F11 and ESC)
- shortcut for new game
- zooming (+/- or Ctrl-+/Ctrl--)

**Comment by AlexErling** [comment]
Hi, I'm new to open source and was wondering if I could take this on. Would you have any suggestions on the best ways to put in shortcuts?

**Comment by nikolockenvitz** [comment]
Hi, of course you can take this. In past I used similar ways like in this line: https://github.com/nikolockenvitz/2048/blob/ad777330fa5677fdf0f89e71bd7f4b207579af45/2048.py#L65, with bind method. But I'm also open to new ideas. For shortcuts regarding zoom etc. you may have to take a look into branch [feature/#1](https://github.com/nikolockenvitz/2048/tree/feature/%231), this is currently in progress, but you can help there too if you like.

If you have any questions feel free to ask...

**Comment by AlexErling** [comment]
I'll start on the full-screen and normal mode/shortcut for new game first.
```
Alexanders-MBP:2048 coding$ python 2048.py
  File "2048.py", line 398
    print(" "*(maxlen-len(str(number)))+str(number), end=" ")
```
I tried running the file and I got this error? What's the best way to get the file to run so I can check my changes?

**Comment by nikolockenvitz** [comment]
Are you using Python 2 or 3? I'm running Python 3.6. I assume that `, end=" "` is not supported with Python 2. For testing you can just comment it, because this is part of the show-function in back end which is not really used.
If you like you can also refactor this function so that it also supports Python 2.

**Comment by AlexErling** [comment]
Okay, I downloaded Python 3.6 and it works. 

For starting a new game, you just want a key bind that would like to new game right?
So up top I'd need to set a key like this. Where would I put the function that assoicates that key press with new game then?
```
KEYS_UP    = ["w", "Up"]
KEYS_RIGHT = ["d", "Right"]
KEYS_DOWN  = ["s", "Down"]
KEYS_LEFT  = ["a", "Left"]
NEW_GAME   = ['F10']
```
Also,
I found an example of how to make tkinter full screen? Should we do something like this? 
```
import tkinter as tk
geom=""

def fullscreen():
    global geom
    geom = root.geometry()
    w = root.winfo_screenwidth()
    h = root.winfo_screenheight()
    root.overrideredirect(True)
    root.geometry('%dx%d+0+0' % (w, h))

def exitfullscreen():
    global geom
    root.overrideredirect(False)
    root.geometry(geom)

root = tk.Tk()
tk.Button(root,text ="Fullscreen", command=fullscreen).pack()
tk.Button(root,text ="Normal", command=exitfullscreen).pack()
root.mainloop()
```

**Comment by AlexErling** [comment]
Okay, never mind I figured it out in a simpler manner. Now I just need to figure out how to center the content when it is full screen
![2048](https://user-images.githubusercontent.com/29783432/39774008-46795698-52bf-11e8-82b8-df177b7ff0fa.png)

**Comment by nikolockenvitz** [comment]
You may take a look in branch https://github.com/nikolockenvitz/2048/tree/feature/%231 which is for automatically adjusting the content (tile, font, ... size)

**Comment by AlexErling** [comment]
```
def toggle_fullscreen(self, event=None):
        self.state = not self.state  # Just toggling the boolean
        self.root.attributes("-fullscreen", self.state)
        self.resize

    def end_fullscreen(self, event=None):
        self.state = False
        self.root.attributes("-fullscreen", False)
        self.resize
```
So I put the resize in the function as well, but for some reason if won't resize correctly. I also noticed that the resize is slightly slow. Do you think there might be a better way to do this?

**Comment by nikolockenvitz** [comment]
I finished #1 and merged #11. So now you can do it a bit more easier I think. It should work by using `self.root.state("zoomed")` and `self.root.state("normal")`.
